lib/main.dart

```diff
@@ -0,0 +1,69 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_localizations/flutter_localizations.dart';
+import 'package:flutter_mentoring/presentation/localization/locale_inherited_widget.dart';
+
+import 'data/todo_store.dart';
+import 'presentation/localization/app_localization.dart';
+import 'presentation/settings_page.dart';
+import 'presentation/todo_page.dart';
+
+void main() {
+  runApp(LocalizedApp());
+}
+
+/// App with list of items and possibility to show tapped item in header.
+/// Changing language is available on the settings page.
+/// If there are no previously saved by user language then system locale is loaded.
+class LocalizedApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final store = TodoStore();
+
+    return LocaleInheritedWidget(
+      child: Builder(
+        builder: (ctx) => StreamBuilder<Locale>(
+            stream: LocaleInheritedWidget.of(ctx).getLocaleStream(),
+            builder: (context, localeStreamSnapshot) => MaterialAppWidget(
+                  locale: localeStreamSnapshot.data,
+                  store: store,
+                )),
+      ),
+    );
+  }
+}
+
+class MaterialAppWidget extends StatelessWidget {
+  const MaterialAppWidget({@required this.locale, @required this.store});
+
+  final Locale locale;
+  final TodoStore store;
```

Comment:
Кстати, если вдруг это у тебя сгенерированный код
Это тоже один из конвеншенов дарта - сначала описываются конструкторы, потом всё остальное.
Просто подметил на всякий случай, полезно сразу привыкать к правилам "экосистемы"

Reply:
Do you mean that constructor should be even before the class instance fields as it is here? Or that the fields should go first, then constructor, then the rest of the methods?

Reply:
Constructor first. It may look weird for the first time, but you will find it more general after you read much code from other applications, flutter and dart sources.